src/nemos/base_class.py

```diff
@@ -172,6 +204,32 @@ class BaseRegressor(Base, abc.ABC):
     scoring the model, simulating responses, and preprocessing data. Concrete classes
     are expected to provide specific implementations of the abstract methods defined here.
 
+    Parameters
+    ----------
+    regularizer :
+        Regularization to use for model optimization. Defines the regularization scheme
+        and related parameters.
+        Default is UnRegularized regression.
+    solver_name :
+        Solver to use for model optimization. Defines the optimization scheme and related parameters.
+        The solver must be an appropriate match for the chosen regularizer.
+        Default is `None`. If no solver specified, one will be chosen based on the regularizer.
+        Please see table below forregularizer/optimizer pairings.
+
+    +---------------+------------------+-------------------------------------------------------------+
+    | Regularizer   | Default Solver   | Available Solvers                                           |
+    +===============+==================+=============================================================+
+    | UnRegularized | GradientDescent  | GradientDescent, BFGS, LBFGS, ScipyMinimize, NonlinearCG,   |
+    |               |                  | ScipyBoundedMinimize, LBFGSB                                |
+    +---------------+------------------+-------------------------------------------------------------+
+    | Ridge         | GradientDescent  | GradientDescent, BFGS, LBFGS, ScipyMinimize, NonlinearCG,   |
```

Comment:
Add proximal gradient to Ridge. The ProximalGradient is equivalent to the GradientDescent for Unregularized, so I don't know if we want to explicitly list it for Unregularized too, but it would also work there